Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs

```diff
@@ -77,14 +98,14 @@ public void Initialize(TelemetryConfiguration configuration)
                 {
                     if (!this.isInitialized)
                     {
-                        this.isInitialized = true;
-
                         this.telemetryClient = new TelemetryClient(configuration);
                         this.telemetryClient.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("exstat:");
 
                         this.metricManager = new MetricManager(this.telemetryClient);
 
                         this.registerAction(this.CalculateStatistics);
+
+                        this.isInitialized = true;
```

Comment:
If this isn't at the end then you have a race condition

---
In reply to: [107222599](https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/370#discussion_r107222599) [](ancestors = 107222599)